src/react/components/pages/editorPage/editorPage.tsx

```diff
@@ -184,8 +191,20 @@ export default class EditorPage extends React.Component<IEditorPageProps, IEdito
      * Called when a tag from footer is clicked
      * @param tag Tag clicked
      */
-    public onTagClicked = (tag: ITag): void => {
-        this.canvas.current.applyTag(tag.name);
+    private onTagClicked = (tag: ITag): void => {
```

Comment:
I like that `lockedTags` is passed into the canvas via props - Is there a reason we're not doing the same for `selectedTag`?  Let's reduce usage of the canvas ref to as little as possible.

Reply:
I struggled with this for a long time. The main piece that was a blocker was how to differentiate changes. For example, if I select `Tag1` to add it to a tag, I can make the change in `componentDidUpdate` no problem. However, if I change my mind and want to remove `Tag1` from the region, I click it again, setting the state to exactly what it was before and is not noticed as a change in the `componentDidUpdate` function. Happy to chat about other possible solutions though

Reply:
We can skip for now and address this later.